org.jacoco.core/src/org/jacoco/core/internal/analysis/ClassAnalyzer.java

```diff
@@ -82,8 +84,18 @@ public void visitEnd() {
 		};
 	}
 
+	/**
+	 * @return <code>true</code> if method should not be analyzed
+	 */
 	// TODO: Use filter hook in future
-	private boolean isMethodFiltered(final int access, final String name) {
+	private boolean isMethodFiltered(final String className,
+			final String superClassName, final int access, final String name,
+			final String desc) {
+		if ("java/lang/Enum".equals(superClassName)
+				&& ("values".equals(name) || name.equals("valueOf") && desc
```

Comment:
@Godin Shouldn't we also check the signature of the values method? For readability I would prefer braces or a separate statement for each method. Maybe also some inline comments what we're going to filter here.